src/com/twitter/intellij/pants/service/PantsCompileOptionsExecutor.java

```diff
@@ -215,6 +215,9 @@ private GeneralCommandLine getPantsExportCommand(final File outputFile, @NotNull
     commandLine.addParameter("--no-quiet");
     if (getOptions().isImportSourceDepsAsJars()) {
       commandLine.addParameter("export-dep-as-jar");
+      if (myResolveSourcesAndDocsForJars) {
```

Comment:
I'm not sure users mean this when they use this settings.
Namely, I'd think right now users expect this setting to download sources for their 3rdparty dependencies, and they'd expect sources to be accessible for their source dependencies by default (as is the status quo).

I'd be more comfortable with adding another toggle saying "I want sources for my source deps", which can even be on the same line, like:
` [ ] Import source deps as jars [ ] make sources accessible`
I feel like we'd want to turn this toggle on by default at the beginning to keep the UX as close to the original as possible, and tell people to maybe disable it in the future (whereas we may not want to toggle `resolveSourcesAnsDocsForJars` on by default ever for 3rdparty deps).

Reply:
Sounds good :)